src/dotnet/Users.Service/UserAvatars.cs

```diff
@@ -68,12 +68,10 @@ public virtual async Task SetDefault(IUserAvatars.SetDefaultCommand command, Can
         if (!avatarId.IsNullOrEmpty()) {
             var avatar = await _userAvatarsBackend.Get(avatarId, cancellationToken).ConfigureAwait(false);
             if (avatar == null || avatar.UserId != user.Id)
-                throw new InvalidOperationException("Invalid AvatarId");
+                throw new InvalidOperationException("Invalid AvatarId.");
         }
 
-        var userProfile = await _userProfilesBackend.Get(user.Id, cancellationToken).ConfigureAwait(false);
-        if (userProfile == null)
-            throw new InvalidOperationException("User profile does not exist");
+        var userProfile = await _userProfilesBackend.Get(user.Id, cancellationToken).Required().ConfigureAwait(false);
```

Comment:
`Required` -> `AssertActive`? It's a front-end service.

Reply:
Disagree, `Required` is a generic extension that works with all nullable results